src/Encryption.php

```diff
@@ -45,8 +45,8 @@ class Encryption
      */
     public function __construct($key, $adapter = null)
     {
-        ($adapter !== null) ? $this->setAdapter($adapter) : $this->setAdapter('openssl');
         $this->setKey($key);
+        ($adapter !== null) ? $this->setAdapter($adapter) : $this->setAdapter('openssl');
```

Comment:
The `setKey` will be executed before setting adapter.

When using the `setAdapter` method, it will also set key directly.

And the Encryption adapter class also sets key, but they don't have any key check.